cartoframes/data/utils.py

```diff
@@ -84,7 +93,17 @@ def _warn_new_geometry_column(df):
 
 
 def _compute_geometry_from_geom(geom):
-    return geom.apply(decode_geometry)
+    first_geom = next(item for item in geom if item is not None)
+    enc_type = detect_encoding_type(first_geom)
+    return geom.apply(lambda g: decode_geometry(g, enc_type))
+
+
+def _first_value(array):
+    array = array.loc[~array.isnull()]  # Remove null values
+    if len(array) > 0:
+        return array.iloc[0]
+    else:
+        warn('Dataset with null geometries')
```

Comment:
This should be a more generic message, since the method is agnostic to the column (it could be used the geometry column or any other column)

Reply:
This comment is still valid for the method in the `Dataset` class

Reply:
What do you suggest here?

Reply:
For this cases I'd prefer to raise an error and let the caller decide to pass or log the error.

Reply:
:+1: